### PR TITLE
Bug 1972598: KubeAPI: Adds install retry doc and reorder existing docs

### PR DESCRIPTION
**Pull request by nmagnezi**
# Description

Documentation changes:
1. Use reference URLs, as opposed to the full GitHub URLs. GitHub URLs hardcode the branch, so this is more branch-friendly.
2. Group all Hive integration documents in one place.
3. Add a doc to describe an install retry.

# What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

# How was this code tested?

Please, select one or more if needed:

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

It's recommended to take a few extra minutes to provide more information about
how this code was tested. Here are some questions that may be worth answering:

- Should this PR be tested by the reviewer?
- Is this PR relying on CI for an e2e test run?
- Should this PR be tested in a specific environment?
- Any logs, screenshots, etc that can help with the review process?

# Assignees

Please, add one or two reviewers that could help review this PR. Use `/assign` if you want to assign
this PR directly to someone.

/cc @filanov 
/cc @rollandf 
/cc @danielerez 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [x] Are the title and description (in both PR and commit) meaningful and clear?
- [x] Is there a bug required (and linked) for this change?
- [x] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by jparrill — APPROVED** [review]
LGTM

File: docs/hive-integration/kube-api-install-retry.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Hive Integration - Retry to Install a cluster
+
+In a case where the cluster installation has failed, the user may want to restart the installation.
+A failed install may happen for multiple reasons; make sure you understand the root cause before trying again.
+
+To restart the installation, the user will need to:
+1. Delete the cluster `AgentClusterInstall` resource.
+2. Delete all the cluster `BareMetalHost` resources (a single resource in the case of SNO).
+3. Create the cluster `AgentClusterInstall` resource again.
+4. Create all the cluster `BareMetalHost` resources (a single resource in the case of SNO).
```
Comment (flaper87) [inline docs/hive-integration/kube-api-install-retry.md:10]: What do you think about rephrasing tihs to something like:

Recreate the BMH resources that were deleted in step 2

**Review by flaper87 — CHANGES_REQUESTED** [review]
Looks good to me overall, just a small suggestion.

File: docs/hive-integration/kube-api-install-retry.md (hunk 1 above)
Comment (nmagnezi) [reply docs/hive-integration/kube-api-install-retry.md]: done

**Review by flaper87 — APPROVED** [review]
thanks!

**Comment by nmagnezi** [comment]
> New changes are detected. LGTM label has been removed.

Resolved a minor git conflict with https://github.com/openshift/assisted-service/pull/2083

**Pull request merged**